### PR TITLE
Feature/weighted dist

### PR DESCRIPTION
**Pull request by lucas-daher**
#### What is the purpose of this pull request?
Change AB test parameters to be passed to the router as weights of a uniform distribution. The conversion from beta to uniform is done with the ProbabilityYBeatsAll function.

#### What problem is this solving?
Changes the meaning of the parameters will make the router faster, because the sampling process is going to be simpler.

#### How should this be manually tested?
Linking this version and testing against new of prod versions of the router (for roll out and possible roll back)

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by tergol1** [comment]
What about change the minor as this is not a bug fix:

File: node/utils/mathTools/forBetaDistribution/decisionRule.ts [hunk 1]
```diff
@@ -21,6 +21,71 @@ export function ProbabilityOfOneBeatTwo(a: number, b: number, c: number, d: numb
...
+                CurrX.push(X[j])
+                idxs.push(0)
+            }
+        }
+        curr += ProbabilityAllBeatY(Y, CurrX, idxs, 0)
+        curr *= Math.pow(-1, idxs.length)
+        ret += curr
+    }
+    return {a:ret, b:1}
+}
```
Comment (tergol1) [inline node/utils/mathTools/forBetaDistribution/decisionRule.ts]: **nit** I'm trying to think in a way to made this calculation more like a step-by-step solution of a math problem. This suggestion is for the code between lines 24 and 86.

- I'm ok with `ProbabilityAllBeatY`, really well written code!
- I'm not getting the point of `NextComb` (maybe I'm the problem lol). I could only think about something like the following code to generate an array of combinations with size `p`

```
export function Combinations(set: any[], p: number): any[] {
    let combinations = []
    let current = []
    let tailcombs = []

    if (p > set.length || p <= 0) {
        return []
    }

    if (p === set.length) {
        return [set]
    }

    if (p === 1) {
        combinations = []
        for (const elem of set) {
            combinations.push(elem)
        }
        return combinations
    }

    for (let i = 0; i < set.length - p + 1; i++) {
        current = set.slice(i, i + 1)
        tailcombs = Combinations(set.slice(i + 1), p - 1)
        for (const elem of tailcombs) {
            combinations.push(current.concat(elem))
        }
    }
    return combinations
}
```
 With that code, we could execute for each combination (inside a `for` loop varying the quantity `p`) `ProbabilityAllBeatY `.

Explaining what I didn't understand:

```
Recursively NextComb applied to [0, 0, 0]
[0, 0, 0] -> [1, 0, 0] -> [2, 1, 0] -> [3, 1, 0] -> [4, 1, 0] -> [5, 1, 0] -> [6, 1, 0] -> ...

Comment (lucas-daher) [reply node/utils/mathTools/forBetaDistribution/decisionRule.ts]: About NextComb, it actually goes like this:

[0, 0, 0] -> [1, 0, 0] -> [0, 1, 0] -> [1, 1, 0] -> [0, 0, 1] -> [1, 0, 1] -> [1, 1, 1]

and is hopefully (and guaranteed by an invariant) to never be called after this.

I change the i++ to the next line to make it clearer.

File: node/typings/testingParameters.ts [hunk 2]
```diff
@@ -29,11 +30,19 @@ export default class TestingParameters {
     }
 
     public Set = (workspacesData: Map<string, WorkspaceData>) => {
+        let names = Array<string>(0)
+        let beta_params = Array<ABTestParameters>(0)
```
[the remark this answers is not in the document]
Comment (lucas-daher) [reply node/typings/testingParameters.ts]: I'm trying to remember to always use camelCase, but whenever I'm not paying attention, I'll just default back to snake_case.

File: node/abTest/updateParameters.ts [hunk 3]
```diff
@@ -19,12 +19,12 @@ export async function UpdateParameters(account: string, aBTestBeginning: string,
...
-        
         const tsmap = new TSMap<string, ABTestParameters>([...testingParameters.Get()])
-        await router.setParameters(account, {
+        router.setParameters(account, {
             Id: testId,
             parameterPerWorkspace: tsmap,
         })
+        
         return
     }
```
Comment (lucas-daher) [inline node/abTest/updateParameters.ts]: @tergol1 I couldn't figure out how to comment in lines under this one, but in line 40 maybe we should remove the `await`

File: node/utils/mathTools/forBetaDistribution/decisionRule.ts (hunk 1 above)
Comment (tergol1) [reply node/utils/mathTools/forBetaDistribution/decisionRule.ts]: I just saw the order of `++` and now I see that I thought about the algorithm as `++i`, but you actually implemented `i++`. Now I'm ok with both solutions, separating or not the index increment.

File: node/abTest/updateParameters.ts (hunk 3 above)
Comment (tergol1) [reply node/abTest/updateParameters.ts]: Yes, we should! Great remark!

**Pull request merged**